### DISCUSSION TITLE
How to install the plugin vim-lastplace?

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#1925] [Q&A] [answered]
Cant install the plugin [VimLastPlace](https://github.com/farmergreg/vim-lastplace). 

Neither use the installatin method `git clone ` (said on the repo)
nor in the **LazyVim** way. So how to install it?

**Comment by dpetka2001** [comment]
Lazyvim already has a `last_loc` [autocmd](https://github.com/LazyVim/LazyVim/blob/e5babf289c5ccd91bcd068bfc623335eb76cbc1f/lua/lazyvim/config/autocmds.lua#L31-L47). Does this not work for you? Does this plugin offer something more? I tried the following by creating a file in `/lua/plugins/lastplace.lua` with the following contents
```lua
return {
  {
    "farmergreg/vim-lastplace",
  },
}
```
and the plugin installed correctly. I don't know how it works, so I didn't test anything else.

**Reply by OriLiMu** [reply]
I want to open the nvim with last session auto saved. 
for example  I open three buffer when I close vim. When I reenter it the same 3 buffers opened how to make it? 
How to make it? 
I use vim last place before. But now seems not worked on lazyvim

**Reply by dpetka2001** [reply]
If you press `s` from the dashboard, then it will restore your last autosaved session including the buffers you had open when you closed neovim.

**Reply by OriLiMu** [reply]
Anyway directly jump to last session without press <kbd>s</kbd>?

**Reply by dpetka2001** [reply]
I don't think so, because sometimes you may not want to restore the last session and just start a fresh one. So, the keymap is there for whenever you want to restore your last session.

**Comment by OriLiMu** [comment]
> I don't think so, because sometimes you may not want to restore the last session and just start a fresh one. So, the keymap is there for whenever you want to restore your last session.

Is there a way to set some command line option to directly open the last session when enter nvim. 
Like `nvim --open-last-session` something

**Reply by dpetka2001** [reply, marked as answer]
You could try `nvim -c "lua require('persistence').load()"` which would load the session for the current directory or `nvim -c "lua require('persistence').load({last=true})"` to load the last session regardless of directory.

**Comment by OriLiMu** [comment]
> You could try `nvim -c "lua require('persistence').load()"` which would load the session for the current directory or `nvim -c "lua require('persistence').load({last=true})"` to load the last session regardless of directory.

One more question plz I tried both of your command looks same what is the meaning of directory here, even the file in the session not in current directory. I check the `persistence` repo at github which is not very active I think I could get some help from you quickly.

**Reply by dpetka2001** [reply]
`persistense` makes use of `mksession` which stores sessions according to the directory you're in. So, if you're in `~/.config/nvim` directory and open the file `/lua/config/keymaps.lua` that will be stored in the session. Now you go to another directory, let's say `~/.config/kitty` and you open the file `kitty.conf`. This will be stored seperately as a session for the current directory. The command `nvim -c "lua require('persistence').load()"` should load the session according to which directory you're in. When you also set `last=true`, then it will just load the last session you were in regardless of the current directory you're in.
